tools/lib/h5tools_utils.c

```diff
@@ -196,7 +196,7 @@ get_option(int argc, const char **argv, const char *opts, const struct long_opti
         /* long command line option */
         int        i;
         const char ch      = '=';
-        char *     arg     = &argv[opt_ind][2];
+        char *     arg     = (char *)&argv[opt_ind][2]; /* TODO: should not cast away const! */
```

Comment:
Looks like you can strdup at the top of the block, `char * arg = strdup(&argv[opt_ind][2]);`, `free(arg)` at the bottom, and then replace occurrences of `argv[opt_ind][2]` with `arg`.

Reply:
Good idea, done.